ci/test-sriov-secondary-network-aws.sh

```diff
@@ -0,0 +1,554 @@
+
+    gateway=$(echo $NETWORK | awk -F'.' '{print $1 "." $2 "." $3 "." $4+1}')
+
+    # Create IP Pool YAML file
+    echo "Generating IPPool YAML..."
+    cat << EOF > "$IP_POOL_YAML"
+apiVersion: crd.antrea.io/v1beta1
+kind: IPPool
+metadata:
+  name: pool1
+spec:
+  ipRanges:
+  - cidr: $ip_pool_cidr
+  subnetInfo:
+    gateway: $gateway
+    prefixLength: $NETMASK
+EOF
+    echo "IP pool YAML file created at $IP_POOL_YAML."
+    kubectl apply -f $IP_POOL_YAML
+
+     # Create NetworkAttachmentDefinition
+    kubectl apply -f "$NAD_YAML"
+    echo "Created NetworkAttachmentDefinition"
+}
+
+run_test() {
+     kubectl apply -f "$SRIOV_SECONDARY_NETWORKS_YAML"
+     kubectl apply -f https://raw.githubusercontent.com/k8snetworkplumbingwg/sriov-network-device-plugin/refs/heads/master/deployments/sriovdp-daemonset.yaml
+     kubectl get nodes -o go-template='{{range .items}}{{.metadata.name}}{{" "}}{{.status.allocatable}}{{"\n"}}{{end}}'
+     kubectl apply -f https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/raw/master/artifacts/networks-crd.yaml
+     create_ippool_and_network_attachment_definition
+     kubectl taint nodes --all node-role.kubernetes.io/control-plane- || true
+     CONTROLPLANE_NODE=$(kubectl get nodes -l node-role.kubernetes.io/control-plane -o jsonpath='{.items[0].metadata.name}')
+     WORKER_NODE=$(kubectl get nodes -l '!node-role.kubernetes.io/control-plane' -o jsonpath='{.items[0].metadata.name}')
+     kubectl label node "$CONTROLPLANE_NODE" eni-id="$CONTROLPLANE_NODE_ENI"
+     kubectl label node "$WORKER_NODE" eni-id="$WORKER_NODE_ENI"
+
+     go test -v -timeout="$TIMEOUT" antrea.io/antrea/test/e2e-secondary-network -run=TestSRIOVNetwork -provider=remote -remote.sshconfig="$SSH_CONFIG_FILE" -remote.kubeconfig="$KUBECONFIG_FILE" -deploy-antrea=false
+}
+
+clean_up() {
+    set +e
+    INSTANCE_ID=$1
+    ENI_ID=$2
+
+    echo "Terminating EC2 instance: $INSTANCE_ID"
+    aws ec2 terminate-instances --instance-ids "$INSTANCE_ID"
+
+    # Check if the EC2 instance termination was successful
+    if [ $? -eq 0 ]; then
+      echo "Successfully terminated EC2 instance: $INSTANCE_ID"
+    else
+      echo "Failed to terminate EC2 instance: $INSTANCE_ID"
+    fi
+
+    echo "Detaching network interface: $ENI_ID"
+    ATTACHMENT_ID=$(aws ec2 describe-network-interfaces --network-interface-ids "$ENI_ID" --query "NetworkInterfaces[0].Attachment.AttachmentId" --output text)
+
+    if [ "$ATTACHMENT_ID" != "None" ]; then
+      aws ec2 detach-network-interface --attachment-id "$ATTACHMENT_ID"
+      if [ $? -eq 0 ]; then
+        echo "Successfully detached network interface: $ENI_ID"
+      else
+        echo "Failed to detach network interface: $ENI_ID"
+      fi
+    else
+      echo "Network interface is not attached to any instance."
+    fi
+    #After deleting the instance, need to wait for a period of time until the ENI status changes to "Available" before it can be deleted.
+    sleep 100
```

Comment:
Is there a way to check the ENI status instead of waiting?

Reply:
Updated.